### PR TITLE
remove null prepend signature type

### PR DESCRIPTION
**Pull request by BrendanChou**
*REQUIRES JS CHANGE ON SIGNATURE TYPES*

### CODE REVIEW

File: contracts/lib/TypedSignature.sol [hunk 1]
```diff
@@ -31,13 +31,11 @@ pragma experimental "v0.5.0";
...
 library TypedSignature {
 
-    enum SignatureType {
-        INVALID,
-        ECRECOVER_NULL,
-        ECRECOVER_DEC,
-        ECRECOVER_HEX,
-        UNSUPPORTED
-    }
+    // Solidity does not offer guarantees about enum values, so we define them explicitly
```
Comment (AntonioJuliano) [inline contracts/lib/TypedSignature.sol]: Is this true?? If so we should also update the ones in `MarginAdmin`

Comment (AntonioJuliano) [reply contracts/lib/TypedSignature.sol]: / anywhere else we use enums publicly

File: contracts/lib/TypedSignature.sol [hunk 2]
```diff
@@ -31,13 +31,11 @@ pragma experimental "v0.5.0";
...
+    uint8 private constant SIGTYPE_INVALID = 1;
```
Comment (AntonioJuliano) [inline contracts/lib/TypedSignature.sol]: Why do these start 1 indexed?

File: contracts/lib/TypedSignature.sol (hunk 1 above)
Comment (BrendanChou) [reply contracts/lib/TypedSignature.sol]: will open different PR

**Pull request merged**